### PR TITLE
Include build docs for fstore

### PR DESCRIPTION
**Pull request by AdityaAtulTewari**

### CODE REVIEW

**Comment by patrickkenney9801** [comment]
is this complete?

**Comment by AdityaAtulTewari** [comment]
> is this complete?

No I should have made this draft, but basically @anish-palakurthi will give me feedback on it.

File: module/MODULE.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+## Build and Deploy Kernel
+
+This should mostly be handled by `make -C module -j$(nproc) ${MODULE}/lib`.
+However first you need to introduce environment variables that are going to aid this.
+
+```shell
+export SUDO="sudo"
+export MODULE=/
+export KHEADS=/usr/src/linux-headers-6.12.0-gadc218676eef
```
Comment (patrickkenney9801) [inline module/MODULE.md:17]: the 6.12 hash is `adc218676eef25575469234709c2d87185ca223a`, is the `g` intentional?

Comment (AdityaAtulTewari) [reply module/MODULE.md:17]: Pretty sure it stands for generic. Run uname-r on caferri or moriarty.

Comment (patrickkenney9801) [reply module/MODULE.md:17]: that tracks

File: module/MODULE.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+export KBUILD=<some-build-src>
+```
+
+1. Build the Kernel
+2. Build the Modules
+3. Build the Headers
+4. Install Headers
+5. Install Modules
```
Comment (patrickkenney9801) [inline module/MODULE.md:25]: I would clarify which of these steps are handled by `make -C module all`

File: module/MODULE.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# Feature Store Module
+
+## Dependencies
+
```
Comment (patrickkenney9801) [inline module/MODULE.md:4]: I would add a warning about pahole

File: module/MODULE.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+1. Build the Kernel
+2. Build the Modules
+3. Build the Headers
+4. Install Headers
+5. Install Modules
+    - These are handled by `make -C module -$(nproc) ${MODULE}/lib`
+6. Install Kernel
+    - This is done with `sudo make -C ${KBUILD} install`
+
+Then in `/etc/default/grub`,
```
Comment (patrickkenney9801) [inline module/MODULE.md:30]: please add that this only needs to be done once

File: module/MODULE.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+Something like:
+`GRUB_DEFAULT="Advanced options for Ubuntu>Ubuntu, with Linux 4.14.31-041431-generic"`
+
+Then the following is sufficient to ensure things work:
+
+Where `"1>0"` is the short string corresponding to the kernel you want to boot into.
+
+```shell
+sudo grub-reboot "1>0" && sudo reboot
+```
```
Comment (patrickkenney9801) [inline module/MODULE.md:68]: it would not be a bad idea to add your ssh retry bash line here to help out

File: module/MODULE.md (hunk 4 above)
Comment (patrickkenney9801) [reply module/MODULE.md:30]: maybe give this a sepaarte section header too

File: module/MODULE.md (hunk 3 above)
Comment (AdityaAtulTewari) [reply module/MODULE.md:4]: Isn't that in the current thing about what to install?

File: module/MODULE.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+Something like:
+`GRUB_DEFAULT="Advanced options for Ubuntu>Ubuntu, with Linux 4.14.31-041431-generic"`
+
+Then the following is sufficient to ensure things work:
+
+Where `"1>0"` is the short string corresponding to the kernel you want to boot into.
+
+```shell
+sudo grub-reboot "1>0" && sudo reboot
```
Comment (patrickkenney9801) [inline module/MODULE.md:51]: I would separate this into a new section stating `Booting into a new Kernel` to make it easier to fine and make it clear this is not 1 time only

**Review by patrickkenney9801 — APPROVED** [review]

**Pull request merged**